src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
And windows filesystem is case-insensitive but base64 is based on case. This increases collision probability enourmously.